File: injector/src/test/java/org/codejargon/feather/PojoProvidedThroughModuleTest.java
package org.codejargon.feather;

import org.junit.Test;

import static org.junit.Assert.assertNotNull;

public class PojoProvidedThroughModuleTest {
    @Test(expected = InjectorException.class)
    public void pojoNotProvided() {
        Injector injector = Injector.with();
        injector.instance(Pojo.class);
    }

    @Test
    public void pojoProvided() {
        Injector injector = Injector.with(new Module());
        assertNotNull(injector.instance(Pojo.class));
    }

    public static class Module {
        @Provides
        Pojo pojo() {
            return new Pojo("foo");
        }
    }

    public static class Pojo {
        private final String foo;

        public Pojo(String foo) {
            this.foo = foo;
        }
    }
}

File: photon/src/test/kotlin/net/mready/photon/PojoProvidedThroughModuleTest.kt
package net.mready.photon

import org.junit.Assert.assertNotNull
import org.junit.Test

class PojoProvidedThroughModuleTest {
    @Test(expected = InjectorException::class)
    fun pojoNotProvided() {
        val injector = Injector.with()
        injector.instance(Pojo::class.java)
    }

    @Test
    fun pojoProvided() {
        val injector = Injector.with(Module())
        assertNotNull(injector.instance(Pojo::class.java))
    }

    class Module {
        @Provides
        fun pojo(): Pojo {
            return Pojo("foo")
        }
    }

    class Pojo(private val foo: String)
}
